configurator.go

```diff
@@ -73,38 +82,114 @@ func (c *Config) Parse() (*Config, diag.Diagnostics) {
 
 	// default filename to 'config' if not provided.
 	if c.FileName == "" {
-		c.FileName = "config"
+		c.FileName = DEFAULT_CONFIG_FILENAME
 	}
 	c.parsed = nil
 
-	// Process OS application directory configuration files.
-	c, diags = processFileConfig(diags, diag.ComponentGlobalFile, c)
+	// todo: make the following processX functions private methods
+
+	c, diags = processFileFlagConfig(diags, c)
 
-	// Process current working directory configuration files.
-	c, diags = processFileConfig(diags, diag.ComponentLocalFile, c)
+	if c.ConfigFilePath != "" {
+		// Process OS application directory configuration files.
+		c, diags = processFileConfig(diags, diag.ComponentFlagFile, c)
+	} else {
+		// Process OS application directory configuration files.
+		c, diags = processFileConfig(diags, diag.ComponentGlobalFile, c)
 
-	// Process environment variable configuration.
-	c, diags = processConfig(diags, diag.ComponentEnvVar, c, c.Env)
+		// Process current working directory configuration files.
+		c, diags = processFileConfig(diags, diag.ComponentLocalFile, c)
+
+		// Process environment variable configuration.
+		c, diags = processConfig(diags, diag.ComponentEnvVar, c, c.Env)
+	}
 
-	// TODO: flags config provider
-	// Process CLI provided flag arguments.
+	// Process CLI provided flag configuration.
+	if c.Flag != nil {
+		c.Flag.Init()
```

Comment:
this _feels_ like this should be done somewhere else... should parsing be init'ing?